helix-term/src/commands.rs

```diff
@@ -3419,27 +3420,50 @@ fn open(cx: &mut Context, open: Open) {
             (0, 0)
         } else {
             (
-                line_end_char_index(&text, line_num),
+                line_end_char_index(&doc_text, line_num),
                 doc.line_ending.len_chars(),
             )
         };
 
-        let indent = indent::indent_for_newline(
-            doc.language_config(),
-            doc.syntax(),
-            &doc.config.load().indent_heuristic,
-            &doc.indent_style,
-            doc.tab_width(),
-            text,
-            line_num,
-            line_end_index,
-            cursor_line,
-        );
+        let comment_token = {
+            let tokens = doc
+                .language_config()
+                .and_then(|config| config.comment_tokens.as_ref());
+
+            get_comment_token(doc.text(), tokens, cursor_line)
+        };
+        let new_line_will_be_comment = config.continue_comments && comment_token.is_some();
```

Comment:
We can clean this up a bit to both signal that we will be continuing the comment and carry the comment token:

```rust
let continue_comment_token = if config.continue_comments {
    doc.language_config()
        .and_then(|config| config.comment_tokens.as_ref())
        .and_then(|tokens| comment::get_comment_token(text, tokens, cursor_line))
} else {
    None
};
```

then in the `match` below we check if `continue_comment_token.is_some()` instead